File: java/acronym/src/main/java/Acronym.java
import io.vavr.collection.CharSeq;

class Acronym {

    private final CharSeq phrase;

    Acronym(String phrase) {
        this.phrase = CharSeq.of(phrase);
    }

    String get() {
        var acronym = phrase
                .zip(phrase.prepend(' '))
                .filter(l -> l._2 == ' ')
                .map(l -> Character.toUpperCase(l._1))
                .mkString();
        return acronym;
    }

}

File: kotlin/acronym/src/main/kotlin/Acronym.kt
object Acronym {
    fun generate(phrase: String): String {
        return phrase.indices
                .filter { i -> phrase.wordStartsAt(i) }
                .map { i -> phrase[i].toUpperCase() }
                .joinToString("")
    }

    private fun String.wordStartsAt(i: Int): Boolean {
        if (i == 0) {
            return true
        }
        if (this[i - 1] == ' ') {
            return true
        }
        if (this[i].isUpperCase() && this[i - 1].isLowerCase()) {
            return true
        }
        
        return false
    }
}